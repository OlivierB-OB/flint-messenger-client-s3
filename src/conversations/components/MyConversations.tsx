import Box from '@material-ui/core/Box';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
import React from 'react';
import { connect } from 'react-redux';
import { Link, useRouteMatch } from 'react-router-dom';
import { IAppState } from '../../appReducer';
import { IConversation, IConversationsStatus } from '../types';
import { ConversationListItem } from './ConversationListItem';
import { Loading } from '../../layout/components/Loading';

export interface IConversationListProps {
  status: IConversationsStatus;
  list: IConversation[];
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    selected: {
      color: theme.palette.secondary.contrastText,
      backgroundColor: theme.palette.secondary.light,
    },
  }),
);

export function ConversationList({ status, list }: IConversationListProps) {
  const loading = status === 'unavailable' ? <Loading /> : null;
  const { selected } = useStyles();
  const match = useRouteMatch<{ conversationId: string }>('/conversation/:conversationId');
  const conversationId = match?.params.conversationId;
  return (
    <Box style={{ minWidth: '300px' }}>
      {loading}
      <List>
        {loading || list.length ? null : (
          <ListItem>
            <ListItemText primary="no conversation available..." />
          </ListItem>
        )}
        {list.map((conversation) => (
          <ListItem
            className={conversationId === conversation._id ? selected : undefined}
            button
            component={Link}
            to={`/conversation/${conversation._id}`}
            key={conversation._id}
          >
            <ConversationListItem conversation={conversation} />
          </ListItem>
        ))}
      </List>
    </Box>
  );
}

const mapStateToProps = ({ conversations }: IAppState) => ({
  status: conversations.status,
  list: conversations.conversations,
});

export const MyConversations = connect(mapStateToProps)(ConversationList);
