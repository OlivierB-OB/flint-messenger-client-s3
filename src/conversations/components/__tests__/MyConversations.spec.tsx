import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import { MyConversations } from '../MyConversations';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('MyConversations', () => {
  let Connected: any = MyConversations;

  it('should provide the status, conversation list to ConversationList', async () => {
    const store = mockStore({
      conversations: {
        status: 'the status',
        conversations: 'the conversation list',
      },
    });
    const component = shallow(<Connected store={store} />).dive();
    expect(component).toMatchSnapshot();
  });
});
