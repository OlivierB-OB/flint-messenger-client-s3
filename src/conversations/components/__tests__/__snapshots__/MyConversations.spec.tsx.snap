// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyConversations should provide the status, conversation list to ConversationList 1`] = `
<ConversationList
  dispatch={[Function]}
  list="the conversation list"
  status="the status"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
